File: p4/devorador_tiempos.py
import json
import time

from auxiliar import generar_mapa_grafo
from coloreado_grafo import realizar_voraz

if __name__ == "__main__":
    n = 4
    t1 = 0
    t2 = 0
    while n <= 65536:
        mapa = generar_mapa_grafo(n)
        t1 = time.time()
        for i in range(100):
            solucion = realizar_voraz(mapa["grafo"])
        t2 = time.time()
        tiempo = (t2 - t1) * 1000
        print(f"n: {n} Tiempo(ms): {tiempo}")
        n *= 2
        

File: p4/ColoreadoGrafo/src/p4/DevoradorTiempos.java
package p4;

import java.io.FileReader;
import java.util.List;
import java.util.Map;

import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;

public class DevoradorTiempos {
	public static void main(String[] args) {
		
		long t1,t2,tiempo;
		JSONParser parser = new JSONParser();
		
		for (int i= 4; i<65537; i*=2) {
			
			String file = "sols/g" + i + ".json";
			try (FileReader reader = new FileReader(file)) {
				JSONObject jsonObject = (JSONObject) parser.parse(reader);
				@SuppressWarnings("unchecked")
				Map<String, List<String>> grafo = (Map<String, List<String>>) jsonObject.get("grafo");
				
				// Medicion de tiempos del algoritmo
				t1 = System.currentTimeMillis ();
				Map<String, String> solucion;
				for (int nVeces=0; nVeces<100; nVeces++) {		// con 100 repes para mejores tiempos
					solucion = ColoreoGrafo.realizarVoraz(grafo);
				}
				t2 = System.currentTimeMillis ();
				tiempo = t2 - t1;
				System.out.println("n: " + i + " Tiempo(ms): " + tiempo);
				
			} catch (Exception e) {
				e.printStackTrace();
			}
			
		}
	}
}
